src/slicegpt/hf_utils.py

```diff
@@ -102,39 +107,65 @@ def get_model_and_tokenizer(
     model.eval()  # This switches off dropout.
     model_adapter.use_cache = False
 
-    tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=True, token=token, local_files_only=local_model)
+    tokenizer = AutoTokenizer.from_pretrained(model_name, use_fast=True, token=token, local_files_only=local_model)
 
     model_adapter.post_init(tokenizer)
     logging.info("Loading model done")
 
     return model_adapter, tokenizer
 
 
-@do_not_initialize
 def load_sliced_model(
     model_name: str,
     sliced_model_path: str,
     *,
     token: str | None = None,
-    lora_config: LoraConfig = None,
+    lora_config: LoraConfig | None = None,
     sparsity: float | None = None,
     round_interval: int | None = 1,
-) -> tuple[ModelAdapter, PreTrainedTokenizerBase]:
+) -> tuple[ModelAdapter | torch.nn.Module, PreTrainedTokenizerBase]:
     """
     Load the sliced model and the tokenizer from the given path. If lora_config is supplied as an arg then this
     function will return a PEFT model (post-slicing finetuned model).
     The corresponding model adapter class must be imported before calling this method.
     """
-    my_model_suffix = pathlib.Path(model_name).name
-    my_sliced_model_name = f"{my_model_suffix}_{sparsity}.pt"
-    my_sliced_model_config = f"{my_model_suffix}_{sparsity}.json"
 
     model_adapter, tokenizer = get_model_and_tokenizer(
         model_name,
         model_path=sliced_model_path,
         uninitialized=True,
         token=token,
     )
+
+    # handle loading sliced HF compatible models
+    if model_name.startswith("microsoft") or model_name.startswith("llama"):
+        new_embedding_dimension = int((1 - sparsity) * model_adapter.hidden_size)
+        new_embedding_dimension -= new_embedding_dimension % round_interval
+
+        scheduler = ConstSlicingScheduler(new_embedding_dimension)
+
+        layers = model_adapter.get_layers()
+        scheduler.setup(
+            hidden_size=model_adapter.hidden_size,
+            layers_num=len(layers),
+            parallel_blocks=True,
+        )
+
+        sliced_model = SlicedPhiForCausalLM.from_pretrained(
```

Comment:
The `sliced_model` can be of type `SlicedPhiForCausalLM` or `SlicedLlamaForCausalLM` at this point. Debugging..